src/main/java/de/hysky/skyblocker/mixin/HandledScreenMixin.java

```diff
@@ -229,6 +230,9 @@ protected HandledScreenMixin(Text title) {
             }
             return;
         }
+        if (currentSolver instanceof ChocolateFactorySolver chocolateFactorySolver) {
+            chocolateFactorySolver.markDirty();
+        }
```

Comment:
This is going to have some nasty merge conflicts with 1.20.5 so I'll review it after that's merged.